src/core/lib/viewport-transition-utils.js

```diff
@@ -63,14 +68,12 @@ export function areViewportsEqual(startViewport, endViewport) {
 */
 export function viewportLinearInterpolator(startViewport, endViewport, t) {
   const viewport = {};
+  assert(Object.keys(startViewport).length === Object.keys(endViewport).length);
```

Comment:
I think we have a compareObjects utility that the props diffing uses. maybe it could be reused?

Reply:
yeah, we have something similar, but that needs some refactor so it can be used here, also I need to assert differently on each of viewport prop so need to generalize this , may be using a `assertFunction` passed as an argument.